src/components/Ratings/Ratings.tsx

```diff
@@ -0,0 +1,281 @@
+import * as React from 'react';
+import { RadioButton as CoreRadio } from 'wix-ui-core/radio-button';
+import styles from './Ratings.st.css';
+import { ReactComponent as StarIcon } from '../../assets/icons/Star.svg';
+import { RATINGS_DATA_HOOKS, RATINGS_DATA_KEYS } from './dataHooks';
+import classNames from 'classnames';
+
+export enum Size {
+  Small = 'small',
+  Large = 'large',
+}
+
+export enum Mode {
+  Input = 'input',
+  Display = 'display',
+}
+
+export enum Layout {
+  Aside = 'aside',
+  Bottom = 'bottom',
+}
+
+export interface RatingsProps {
+  value: number;
+  mode: Mode;
+  onSelect?(value: number): void;
+  inputOptions?: string[];
+  disabled?: boolean;
+  error?: boolean;
+  size?: Size;
+  layout?: Layout;
+  ratingDisplay?: string;
+  countDisplay?: string;
+  'data-hook'?: string;
+}
+
+interface DefaultProps {
+  value: number;
+  disabled: boolean;
+  error: boolean;
+  mode: Mode;
+  size: Size;
+  layout: Layout;
+  inputOptions: string[];
+}
+
+interface RatingsState {
+  currentHovered: number;
+  currentFocus: number;
+}
+
+// Define values for keycodes
+const VK_LEFT = 'ArrowLeft';
+const VK_UP = 'ArrowUp';
+const VK_RIGHT = 'ArrowRight';
+const VK_DOWN = 'ArrowDown';
+
+/** Ratings component based on IconToggle */
+export class Ratings extends React.Component<RatingsProps, RatingsState> {
+  static displayName = 'Ratings';
+  static defaultProps: DefaultProps = {
+    value: 0,
+    disabled: false,
+    error: false,
+    mode: Mode.Input,
+    size: Size.Small,
+    layout: Layout.Aside,
+    inputOptions: [],
+  };
+
+  iconRefs = [];
+
+  state: RatingsState = {
+    currentHovered: 0,
+    currentFocus: 0,
+  };
+
+  getDataAttributes() {
+    const { disabled, error, size } = this.props;
+
+    return {
+      [RATINGS_DATA_KEYS.Disabled]: disabled,
+      [RATINGS_DATA_KEYS.Error]: error,
+      [RATINGS_DATA_KEYS.Size]: size,
+    };
+  }
+
+  handleHoverIcon = (idx: number) => {
+    this.setState({ currentHovered: idx });
+  };
+
+  handleUnhoverIcon = () => {
+    this.setState({ currentHovered: 0 });
+  };
+
+  handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    const { currentFocus } = this.state;
+
+    if (currentFocus === 0) {
+      return;
+    }
+
+    switch (e.key) {
```

Comment:
It's not really important, but I think I would do something like:

```js
let direction = 0;

switch (key) {
  case UP:
  case RIGHT:
    direction = 1;
  break;
  case DOWN:
  case LEFT:
    direction = -1;
  break;
}

if (direction) {
  let nextFocus = ((currentFocus + direction) % 5);
  nextFocus = nextFocus < 0 ? nextFocus + 5 : nextFocus;
  e.preventDefault();
  this.setState({ currentFocus: nextFocus }, () => {
   this.iconRefs[nextFocus].focus();
  });
  return false;
}
```
This way there's no repetitive code, and the `return` value is in one place.

Reply:
btw, we also need to listen for `SPACE` and `ENTER` in order to select the focused star.
And also, we need to "clean" the `currentFocus` on the root's `blur` and `focus` events.